### PR TITLE
Adopt Relic for release infrastructure

### PR DESCRIPTION
**Pull request by silasdavis**
Imported Relic to control release process and versioning:
- Stores change notes and versions in a slice of structs as a single source of truth to generate runtime change information, print version, generate changelogs, and tag releases
- Enforces that versions numbers advance in semantic versioning increments
- Adds scripts to automate version tagging for releases (actual release triggered by CircleCI)

I didn't pull in Goreleaser for releasing binaries for Burrow just yet - will save that for a future PR when we review how docker image we are pushing as current 'release'

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# burrow changelog
-## v0.17.1
+# Bosmarmot Changelog
```
Comment (compleatang) [inline CHANGELOG.md]: ??? Is this supposed to be here?

File: release/changelog_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+				Notes:   `Wonderful things were achieved`,
+			},
+			{
+				Version: "0.0.1",
+				Notes:   `Marvelous advances were made`,
+			},
+		})
+	assert.NoError(t, err)
+
+	assert.Equal(t, `# Bosmarmot Changelog
```
Comment (compleatang) [inline release/changelog_test.go]: ??? shouldn't this be Burrow?

File: CHANGELOG.md (hunk 1 above)
Comment (silasdavis) [reply CHANGELOG.md]: lol well spotted COPYPASTA!

File: release/changelog_test.go (hunk 2 above)
Comment (silasdavis) [reply release/changelog_test.go]: This one is just a test, but still

**Pull request merged**